### DISCUSSION TITLE
TypeError: Cannot read property 'findMany' of undefined

### DISCUSSION THREAD

**Discussion by ynsss** [prisma/prisma#3057] [Q&A] [answered]
I wanted to make a chat app to practice working with graphql and node, for database I used prisma. I was doing everything like in this tutorial.

https://www.howtographql.com/graphql-js/0-introduction/

I just changed variable names.

so I have this code

```ts
const { GraphQLServer } = require('graphql-yoga')
const { PrismaClient } = require('@prisma/client')
const prisma = new PrismaClient()


const resolvers = {
  Query: {
    history: async (parent, args, context) => {
      return context.prisma.Messages.findMany()
    },
  },
  Mutation: {
    post: (parent, args, context) => {
      const newMessage = context.prisma.Messages.create({
        data: {
          username: args.username,
          message: args.message,
        },
      })
      return newMessage
    },
  },
}

const server = new GraphQLServer({
  typeDefs: './src/schema.graphql',
  resolvers,
  context: {
    prisma,
  }
})
server.start(() => console.log(`Server is running on http://localhost:4000`))
```
as my index.js

this is my schema.prisma

```prisma
datasource db {
  provider = "sqlite"
  url      = "file:./dev.db"
}


generator client {
  provider = "prisma-client-js"
}


model Message {
  id       Int      @id @default(autoincrement())
  sendedAt DateTime @default(now())
  message  String
  username String
}
```

script.js 

```ts
const { PrismaClient } = require("@prisma/client")


const prisma = new PrismaClient()


async function main() {
  const newMessage = await prisma.mes.create({
    data: {
      message: 'Fullstack tutorial for GraphQL',
      username: 'www.howtographql.com',
    },
  })
  const allMessages = await prisma.mes.findMany()
  console.log(allMessages)
}


main()
  .catch(e => {
    throw e
  })
  // 5
  .finally(async () => {
    await prisma.disconnect()
  })
```

and schema.graphql

```graphql
type Query {
  history: [Message!]!
}

type Mutation {
  post(username: String!, message: String!): Message!
}

type Message {
  id: ID!
  message: String!
  username: String!
}
```

and that is what i got in my playground

```{
  "data": null,
  "errors": [
    {
      "message": "Cannot read property 'findMany' of undefined",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "history"
      ]
    }
  ]
}
```

please help

**Comment by ryands17** [comment, marked as answer]
@ynsss It should be `prisma.message.findMany`. It's the lowercase version of the model name you define in your `schema.prisma`.

**Reply by ynsss** [reply]
Thank you very much

**Reply by Allislove** [reply]
Thanks man.

**Reply by bmarianome** [reply]
Hey, i'm having the same error but it is already lowercase. The error just occurs in production (deployed to vercel), in development works fine

**Reply by janpio** [reply]
Make sure to try to deploy with an empty cache on Vercel - sometimes Vercel does not pick up any schema changes and deploys an old Client that might be missing newly added models.

**Reply by AlexandroMtzG** [reply]
Thanks @janpio :)

**Reply by JuanCarlosLeone** [reply]
I'm having the same error but it is already lowercase.

```
 await prismaClient.emergency.create({
            data: {
                telephone: emergencyData.telephone,
                status: emergencyData.status,
                client: emergencyData.client,
            },
        });
```

**Reply by efkann** [reply]
> Make sure to try to deploy with an empty cache on Vercel - sometimes Vercel does not pick up any schema changes and deploys an old Client that might be missing newly added models.

This happened on Vercel to me, and skipping the build cache worked 🙏  Also, you can now set an environment variable `VERCEL_FORCE_NO_BUILD_CACHE` to true or redeploy from the dashboard. 

More here : https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#managing-build-cache

**Reply by LukaHietala** [reply]
I had the same issue in my local environment, but I managed to fix it just by deleting the `.next` folder which stores all the cache and few other things and restarting the dev server 😅

**Reply by ghost** [reply]
Thank you! @LukaHietala

**Reply by Armadillidiid** [reply]
You're life saver! @LukaHietala

**Reply by GAMEA9G** [reply]
if that does not solve the issue
change this
generator client {
  provider = "prisma-client-js"
  output   = "../generated/prisma"
}
to this
generator client {
  provider = "prisma-client-js"
  output   = "../node_modules/.prisma/client"
}

**Comment by timmeade** [comment]
i just had this same issue:  Happened when pnpm updated

next.js 
turborepo
pnpm
etc

I had to delete the .next folder for that app and the node_modules for that app and the root node_modules

Restarted so everything regenerated.   

All good.

**Reply by sami12389** [reply]
This worked for me. Thanks man!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by GAMEA9G** [comment]
change this
`generator client {
provider = "prisma-client-js"
output = "../generated/prisma"
}`
to this
`generator client {
provider = "prisma-client-js"
output = "../node_modules/.prisma/client"
}`
